src/os/shared.rs

```diff
@@ -34,17 +34,17 @@ impl Iterator for KeyboardEvents {
 
 fn get_datalink_channel(
     interface: &NetworkInterface,
-) -> Result<Box<dyn DataLinkReceiver>, std::io::Error> {
+) -> Result<Box<dyn DataLinkReceiver>, MyError> {
     let mut config = Config::default();
     config.read_timeout = Some(time::Duration::new(1, 0));
 
     match datalink::channel(interface, config) {
         Ok(Ethernet(_tx, rx)) => Ok(rx),
-        Ok(_) => Err(std::io::Error::new(
-            ErrorKind::Other,
-            "Unsupported interface type",
-        )),
-        Err(e) => Err(e),
+        Ok(_) => {
+            let error = MyError::new(MyErrorKind::PermissionError("Please do something".to_string()));
+            Err(error)
+        },
+        Err(e) => Err(MyError::new(MyErrorKind::OtherError(e.to_string()))),
```

Comment:
The permission error actually happens here. You can test for it locally by running without `sudo` and seeing where the error comes from.